### ISSUE TITLE
astro-env breaks PostCSS/Tailwind integration (buildtime)

### ISSUE THREAD

**Issue by Xevion** [florian-lefebvre/astro-env#5]
I tried to use the integration, but I am unsure of why it would cause PostCSS/Tailwind to fail.

You can see the full codebase [here](https://github.com/Xevion/undefined.behavio.rs/tree/astro-env). Copied directly from the README.

Sorry if it's something obvious, my brain may be pretty fried right now.

See config [here](https://github.com/Xevion/undefined.behavio.rs/blob/astro-env/astro.config.mjs#L34), relevant code pasted below.

```ts
export default defineConfig({
  site: "https://undefined.behavio.rs",
  prefetch: true,
  integrations: [tailwind(), sitemap(), robotsTxt(), mdx(), react(), env({
    schema: z.object({
      PUBLIC_VERCEL_GIT_REPO_OWNER: z.string(),
      PUBLIC_VERCEL_GIT_REPO_SLUG: z.string(),
      PUBLIC_VERCEL_GIT_COMMIT_SHA: z.string(),
    }),
  })],
  markdown: {
    syntaxHighlight: false,
    rehypePlugins: [[rehypePrettyCode, prettyOptions]]
  }
});
```

Build error (from Vercel, it already has environment variables):

```
[16:43:56.329] Running build in Washington, D.C., USA (East) – iad1
[16:43:56.518] Cloning github.com/Xevion/undefined.behavio.rs (Branch: astro-env, Commit: af29f08)
[16:43:56.910] Cloning completed: 390.573ms
[16:43:59.327] Restored build cache
[16:43:59.441] Running "vercel build"
[16:43:59.998] Vercel CLI 34.2.2
[16:44:00.834] Detected `pnpm-lock.yaml` version 9 generated by pnpm 9
[16:44:00.848] Installing dependencies...
[16:44:02.290] Lockfile is up to date, resolution step is skipped
[16:44:02.409] Progress: resolved 1, reused 0, downloaded 0, added 0
[16:44:02.490] Packages: +20 -43
[16:44:02.491] ++++++++++++++++++++-------------------------------------------
[16:44:03.396] Progress: resolved 20, reused 0, downloaded 20, added 20, done
[16:44:04.124] 
[16:44:04.125] dependencies:
[16:44:04.125] + astro-env 0.0.5
[16:44:04.125] 
[16:44:04.168] Done in 2.8s
[16:44:04.223] Running "pnpm run build"
[16:44:04.744] 
[16:44:04.744] > @0.0.1 build /vercel/path0
[16:44:04.744] > astro check && astro build
[16:44:04.744] 
[16:44:08.695] node:internal/process/promises:288
[16:44:08.696]             triggerUncaughtException(err, true /* fromPromise */);
[16:44:08.696]             ^
[16:44:08.696] 
[16:44:08.697] [Failed to load PostCSS config: Failed to load PostCSS config (searchPath: /vercel/path0): [Failed to load PostCSS config] Failed to load PostCSS config (searchPath: /vercel/path0): [ReferenceError] module is not defined in ES module scope
[16:44:08.697] This file is being treated as an ES module because it has a '.js' file extension and '/vercel/path0/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
[16:44:08.697] ReferenceError: module is not defined in ES module scope
[16:44:08.697] This file is being treated as an ES module because it has a '.js' file extension and '/vercel/path0/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
[16:44:08.697]     at file:///vercel/path0/postcss.config.js:1:1
[16:44:08.697]     at ModuleJob.run (node:internal/modules/esm/module_job:195:25)
[16:44:08.697]     at async ModuleLoader.import (node:internal/modules/esm/loader:336:24)
[16:44:08.697]     at async importDefault (file:///vercel/path0/node_modules/.pnpm/vite@4.5.0_sass@1.69.5/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:36556:18)
[16:44:08.697]     at async Object.search (file:///vercel/path0/node_modules/.pnpm/vite@4.5.0_sass@1.69.5/node_modules/vite/dist/node/chunks/dep-bb8a8339.js:28992:38)
[16:44:08.697] ]
[16:44:08.698] 
[16:44:08.698] Node.js v18.19.1
[16:44:08.716]  ELIFECYCLE  Command failed with exit code 1.
[16:44:08.732] Error: Command "pnpm run build" exited with 1
[16:44:09.862] 
```

**Comment by florian-lefebvre** [comment]
I also had issues recently but I didn't take the time to fix them, intentionally. Indeed, Astro Env will be released as an experimental feature of Astro 4.10. So this integration will soon be deprecated, sorry about the troubles!

**Comment by florian-lefebvre** [comment]
I'm closing as this package is now deprecated. Astro 4.10 now has `astro:env` in experimental: https://astro.build/blog/astro-4100/
